STACpopulator/extensions/file.py

```diff
@@ -0,0 +1,65 @@
+"""FileHelper module."""
+
+import functools
+from typing import Dict, TypeVar
+
+import pystac
+import requests
+from pystac.extensions.file import FileExtension
+
+from STACpopulator.extensions.base import ExtensionHelper
+
+# Constants
+T = TypeVar("T", pystac.Asset, pystac.Link)
+HTTP_SERVER_ASSET_KEY = "HTTPServer"
+OPEN_DAP_ASSET_KEY = "OpenDAP"
+
+
+class FileHelper(ExtensionHelper):
+    """Helper to handle file info from elements of types Asset and Link."""
+
+    access_urls: Dict[str, str]
+    _session = requests.Session()
```

Comment:
The helper should allow initialization with additional request sessions parameters (see https://github.com/crim-ca/stac-populator/blob/master/STACpopulator/request_utils.py), in case the files are protected or require extra request-specific adjustments.

When invoked in the pipeline, it should pass down the attributes obtained from the upstream CLI call to this helper's creation.  See for example:

https://github.com/crim-ca/stac-populator/blob/c2068a29e222463a1e09e37d4a38215cb2cb3089/STACpopulator/cli.py#L78-L83 
https://github.com/crim-ca/stac-populator/blob/c2068a29e222463a1e09e37d4a38215cb2cb3089/STACpopulator/implementations/CORDEXCMIP6_Ouranos/add_CORDEX6.py#L49-L65